### DISCUSSION TITLE
Weekly Brioche 2/27 - 3/8

### DISCUSSION THREAD

**Discussion by mgatelabs** [mgatelabs/GenshinSolvers#82] [General]
Need to start up a new discussion.

But with over 40 broken puzzles, need to enhance the Three D view to display some static, think I can find a static gif generator. #80

**Comment by mgatelabs** [comment]
Added 4 more background images, bringing the total to 13 Background images.  Still need to setup the JSON.

**Comment by mgatelabs** [comment]
Thinking, need to resolve this Safari issue, there are a lot of iPhones and the way it breaks is just blank.

**Comment by mgatelabs** [comment]
Two more puzzles are online. The main island has 8 puzzles left.

**Comment by mgatelabs** [comment]
@stefnotch for puzzle x16786, you can shrink the ortho value, that will make the pieces larger and then they would better cover up the image pieces.

**Reply by stefnotch** [reply]
Good to know :)
Quick heads up: University has started for me, so I won't have much time in the next few days.

**Reply by mgatelabs** [reply]
Focus on what is really important

**Comment by mgatelabs** [comment]
Thinking I could add another layer to the cubes, so the light up cubes can have some rotation, they would still face the screen, but be tilted.

**Reply by stefnotch** [reply]
As in, allowing the cubes to have a "roll" in addition to a position? That would certainly be quite useful.
![image](https://user-images.githubusercontent.com/10220080/156920967-ff38e242-f029-4fac-98c0-75f2772bd3eb.png)

I wonder if it would be useful to have control over all 3 rotations (pitch, yaw, roll). As long as we're careful to not mess with them too much, it should be quite fine. 🤔

**Reply by mgatelabs** [reply]
I should rewrite the light up cubes and just make them a single face, and use shaders to change the textures, so It could rotate freely, but some of the light up cubes are a mess, so not being 100% accurate makes it easier.

**Comment by mgatelabs** [comment]
Three isn't the safari problem, it's three interactive.

**Reply by stefnotch** [reply]
That's strange, could three.interactive be using some syntax that safari doesn't understand?
Could you try adding this file https://github.com/markuslerner/THREE.Interactive/blob/master/src/index.ts into the project and comment out the three.interactive import? Make sure to reference and use it in some way, even if it's just `console.log(InteractiveObject)`. If it still causes the issue, we now have a chance of figuring out which part of the file causes the issue.

**Reply by stefnotch** [reply]
After taking a closer look at three.interactive, I figured out one of the problems. My bad. https://github.com/markuslerner/THREE.Interactive/pull/11

**Reply by mgatelabs** [reply]
I tried to import each part of the class (from the dependency) one at a time and it all ended the same.

console.log(InteractiveObject); Crash

console.log(InteractiveEvent); Crash

console.log(InteractionManager); Crash

**Reply by mgatelabs** [reply]
Wait, I tried to import the file locally and it's working...

Was able to bring back all the code and it works.  Could there be a problem with how the bundled code is optimized?  It is only a problem for non-dev builds.

I was also able to console.log each of the classes in the class file, it still works.

**Reply by stefnotch** [reply]
Possibly. I guess the easiest solution for now is importing the file locally (and including the license as a comment in the file)

**Reply by mgatelabs** [reply]
Did it and cleaned up the homepage.  Guess it's ready for the masses to play with it.

Could break out the mic and make a how to use video.

**Reply by stefnotch** [reply]
Yes, it's definitely ready! All the important parts are done and it's quite polished :) Now just a ton of puzzles are missing, but that'll happen over time.

Before releasing it, it might make sense to edit https://github.com/mgatelabs/GenshinSolvers#location_cubesjson . Additionally, adding some extra stuff like "what sort of screenshots are good" could be helpful.

**Reply by mgatelabs** [reply]
I'm going to update the wiki pages, so it include the needed info

**Comment by mgatelabs** [comment]
Posted a link to [Reddit](https://www.reddit.com/r/Genshin_Impact/comments/t95x9q/cube_puzzle_solver_webapp_wip/), but art seems to always win.

**Reply by stefnotch** [reply]
I see. Maybe a short video of that one cursed puzzle in the ruins might work?
Preferably with a meme-friendly title like "Drove me insane, brought out the big guns". Or "I will have order!".

![image of the hard puzzle](https://camo.githubusercontent.com/36e3fdfb3335eaf8b1332aab5a0f903493f982d3089531ab5423a3eb5feaedd8/68747470733a2f2f63646e2e646973636f72646170702e636f6d2f6174746163686d656e74732f3930393430373935353734373734353739332f3932373937373033353736313037303131302f756e6b6e6f776e2e706e67)

**Comment by mgatelabs** [comment]
39 Puzzles left, the joy.

Going to try to finish off the 1st island, since I already uploaded images for each puzzle.
